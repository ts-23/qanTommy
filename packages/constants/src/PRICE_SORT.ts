import type { StringKeyOf } from "@repo/types";

export const PRICE_SORT = {
  ASC: "ASC",
  DSC: "DSC",
} as const;

export type PriceSortKey = StringKeyOf<typeof PRICE_SORT>;
