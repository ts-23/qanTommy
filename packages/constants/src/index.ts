export * from "./PRICE_SORT";
