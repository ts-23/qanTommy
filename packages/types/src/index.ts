export * from "./PropertyOffer";
export * from "./PriceSort";
export * from "./StringKeyOf";
