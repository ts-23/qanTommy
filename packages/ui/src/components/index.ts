export * from "./Card";
export * from "./Spinner";
export * from "./Rating";
