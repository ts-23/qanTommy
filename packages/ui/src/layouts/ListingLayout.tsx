import type { PropsWithChildren } from "react";

type ListingLayoutItem = {
  // imageURL?: string;
  // title?: string;
  // subtitle?: string;
  // link?: string;
};

type ListingLayoutProps = { items: ListingLayoutItem[] } & PropsWithChildren;

export function ListingLayout(props: ListingLayoutProps) {
  return (
    <div className="flex h-32">
      <div className="bg-red-700 max-w-2xs flex-1">01</div>
      <div className="bg-green-700 flex-1">2</div>
    </div>
  );
}
