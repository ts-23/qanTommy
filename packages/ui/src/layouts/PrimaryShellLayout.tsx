import type { PropsWithChildren } from "react";

type PrimaryShellLayoutProps = { logoUrl?: string } & PropsWithChildren;

export function PrimaryShellLayout(props: PrimaryShellLayoutProps) {
  return (
    <div className="p-8">
      <div
        className="w-full h-24  bg-red-500"
        style={{
          backgroundImage: `url(${props.logoUrl})`,
        }}
      />
      {props.children}
    </div>
  );
}
