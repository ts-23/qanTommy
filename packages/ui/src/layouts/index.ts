export * from "./PrimaryShellLayout";
export * from "./ListingLayout";
