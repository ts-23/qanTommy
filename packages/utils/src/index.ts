export * from "./startCase";
export * from "./constructString";
