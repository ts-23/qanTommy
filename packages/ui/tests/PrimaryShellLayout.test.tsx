import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { PrimaryShellLayout } from "../src/layouts";

// using toMatchSnapshot to save assessment time. We can cover more test cases and go into as much detail as we want in actual app.
describe("PrimaryShellLayout", () => {
  it("renders with logoUrl", () => {
    const actual = render(<PrimaryShellLayout logoUrl="example.com" />);
    expect(actual.baseElement).toMatchSnapshot();
  });

  it("renders without logoUrl", () => {
    const actual = render(<PrimaryShellLayout />);
    expect(actual.baseElement).toMatchSnapshot();
  });

  it("renders with children", () => {
    const actual = render(
      <PrimaryShellLayout>
        <div>foobar</div>
      </PrimaryShellLayout>
    );
    expect(actual.baseElement).toMatchSnapshot();
  });
});
