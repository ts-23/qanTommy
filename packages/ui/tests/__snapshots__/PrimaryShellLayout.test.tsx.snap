// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrimaryShellLayout > renders with children 1`] = `
<body>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(example.com);"
      />
    </div>
  </div>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(undefined);"
      />
    </div>
  </div>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(undefined);"
      />
      <div>
        foobar
      </div>
    </div>
  </div>
</body>
`;

exports[`PrimaryShellLayout > renders with logoUrl 1`] = `
<body>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(example.com);"
      />
    </div>
  </div>
</body>
`;

exports[`PrimaryShellLayout > renders without logoUrl 1`] = `
<body>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(example.com);"
      />
    </div>
  </div>
  <div>
    <div
      class="p-8"
    >
      <div
        class="w-full h-24 bg-no-repeat"
        style="background-image: url(undefined);"
      />
    </div>
  </div>
</body>
`;
